migrations/20210612_initial.sql

```diff
@@ -1,21 +1,29 @@
 CREATE TABLE contact_profiles ( -- remote user profile
   contact_profile_id INTEGER PRIMARY KEY,
   contact_ref TEXT NOT NULL, -- contact name set by remote user (not unique), this name must not contain spaces
-  display_name TEXT NOT NULL DEFAULT '',
+  display_name TEXT NOT NULL,
   properties TEXT NOT NULL DEFAULT '{}' -- JSON with contact profile properties
 );
 
--- the first record (id = 1) is reserved for the first local user
-INSERT INTO contact_profiles (contact_profile_id, contact_ref) VALUES (1, '');
-
-
 CREATE TABLE users (
   user_id INTEGER PRIMARY KEY,
-  contact_profile_id INTEGER NOT NULL UNIQUE REFERENCES contact_profiles -- user's profile
+  contact_id INTEGER NOT NULL UNIQUE REFERENCES contacts ON DELETE CASCADE
+    DEFERRABLE INITIALLY DEFERRED,
+  active_user INTEGER -- 1 for active user
 );
 
--- the first record (id = 1) is reserved for the first local user
-INSERT INTO users (user_id, contact_profile_id) VALUES (1, 1);
+CREATE TABLE contacts (
+  contact_id INTEGER PRIMARY KEY,
+  contact_profile_id INTEGER UNIQUE REFERENCES contact_profiles, -- NULL if it's an incognito profile
+  local_contact_ref TEXT NOT NULL,
+  lcr_base TEXT NOT NULL,
+  lcr_suffix INTEGER NOT NULL DEFAULT 0,
+  user_id INTEGER NOT NULL REFERENCES users,
+  user INTEGER, -- 1 if this contact is a user
+  created_at TEXT NOT NULL DEFAULT (datetime('now')),
+  UNIQUE (user_id, local_contact_ref) ON CONFLICT FAIL,
```

Comment:
what does `ON CONFLICT FAIL` achieve here? reading up on it, it seems somewhat destructive

Reply:
it allows to stay within the transaction when name clash without rolling back the previous changes - the retry loop that increments the suffix relies on it

Reply:
I didn't test how it works though yet